src/app/shared/services/skin/skin.service.ts

```diff
@@ -35,23 +37,23 @@ export class SkinService extends SyncServiceBase {
       this.appConfigService,
       this.themeService,
       this.templateService,
+      this.appConfigService,
     ]);
-    const skinsConfig = this.appConfigService.APP_CONFIG.APP_SKINS;
-    this.availableSkins = arrayToHashmap(skinsConfig.available, "name");
+    this.subscribeToAppConfigChanges();
```

Comment:
nit(non-blocking)
As we are making changes to the service it might be worthwhile refactoring to register the action handler here also and remove from the `template-action.service`. This would also require adding similar handlers for `emit` functionality as the rest of actions (i.e. a way to register emit handler), or re-authoring to replace emit `set_skin` with a pure `skin set` action

Reply:
Good suggestions. 
Perhaps it's worth some brief discussion about which option would be preferred. Personally I think I prefer the `skin: set` syntax to `emit: set_skin`. But this would be a breaking change for some existing templates, just 2 from ParentApp, plus our debug sheets and a couple more across different deployments.
<img width="300" alt="Screenshot 2023-06-20 at 15 30 38" src="https://github.com/IDEMSInternational/parenting-app-ui/assets/64838927/01fd8af3-e40a-4e3c-a042-b97f06e9fab5">

There are also other actions that use the `emit: ` pattern that would be candidates for being refactored. I'm thinking in particular of `set_theme`, although my main reason for thinking of this over `set_language` is that the latter has its own service, which shouldn't necessarily hold any significance from an authoring perspective
<img width="500" alt="Screenshot 2023-06-20 at 15 40 03" src="https://github.com/IDEMSInternational/parenting-app-ui/assets/64838927/ea4cc7e0-c5c9-4f85-a0f7-b2ad2f52e01a">

So maybe the simplest option would be to add a way to register an emit handler, as you suggest, unless we think it's important to move towards authoring actions in the `skin: set` style

Reply:
Yeah I think now that we have a cleaner system for registering actions the whole `emit` system seems a little unnecessary. 

As an intermediate we could add a small script to the parser to deprecate and rewrite emit statements whilst waiting for authors to update (like how the default parser manages `DEPRECATED_FIELD_NAMES`). Only thing to resolve would be what prefix to use as an action handler`force_reprocess`, `force_reload` etc.... maybe something like `template reload` or `page reload`?

Not super important, but might be good to make a low priority issue to follow-up now this PR has been merged

Reply:
https://github.com/IDEMSInternational/parenting-app-ui/issues/1963